templates/datadog.yaml.j2

```diff
@@ -1,7 +1,14 @@
 # Managed by Ansible
 
-{% if datadog_config["dd_url"] is not defined -%}
-dd_url: {{ datadog_url | default('https://app.datadoghq.com') }}
+{% if datadog_config["dd_url"] is not defined
+   and datadog_url is not defined
+   and datadog_config["site"] is not defined -%}
+site: {{ datadog_site | default('datadoghq.com') }}
```

Comment:
I'd let the Agent define the default value as well, i.e. if the ansible variable is not set, I'm in favor of not setting the option at all in this config.